### PR TITLE
delete-comments

### PR DESCRIPTION
**Pull request by josh8551021**
Added 'delete comments' feature. First commit: f0839b8

### CODE REVIEW

**Comment by josh8551021** [comment]
I just realized that it seems that I never made a pull request for my branch "comments-additional-features". Sorry about that. All of these PRs get really confusing.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (tzsebe) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: In order to code more defensively, you should also define a max number of comments, so a crafty user can't send the service custom javascript to load all the contents and potentially DoS it. (pretend you have millions of comments :-) ).

**Comment by josh8551021** [comment]
So the first commit in this PR is actually faae094.

File: portfolio/src/main/webapp/script.js
Comment (tzsebe) [inline portfolio/src/main/webapp/script.js]: Constructing URL parameters is one of those things you should leave to internal libraries as much as possible. At the very lease, you need to encode the field values so things like spaces and special characters get serialized properly (using `encodeURIComponent`).

Ideally, there's a library that does all the work for you. It looks like there might be a `URLSearchParams` you can use; I'm not sure if this is standard/core javascript, though. I leave it up to you to research that. Here's a starting point: https://stackoverflow.com/questions/111529/how-to-create-query-parameters-in-javascript

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java
Comment (tzsebe) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:22]: Ok this is the question you had during standup, so will address it here.

First off, for THIS review, and in accordance with the [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)), I'll let you get away with it. You can leave this as-is and not worry about it for this PR.

Now... if this were a more involved project, here's what I would have you do (again, you do not need to do this):

* Create a separate class that knows how to talk to your datastore and return entities.
* Have your two servlets depend on that class, rather than directly on the data store.

So you would have a class called `CommentsDataAccessor` (or some name like that), which would have a `DataStoreService` as a member, and it would have `getComments` and `deleteAllComments` methods. It would have other stuff too.

There are a lot of advantages to having something like this. Mainly, it creates much stronger separation of concerns. The Servlet should only care about basic request validation and communicating between with the client.

The CommentsDataAccessor should only care about what it means for there to be "comments" and how to access them.

**Review by tzsebe — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (josh8551021) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: Can I ask what "DoS" means?

Comment (josh8551021) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: Done.

Comment (tzsebe) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: Denial of Service (you often see it as DDoS these days - Distributed....) Basically some nefarious villain trying to take down your site with requests.

**Review by tzsebe — APPROVED** [review]
Nice work on this week's stuff! Approving to unblock; you should change that javascript url encoding thing I mentioned at some point, though it isn't urgent.

File: portfolio/src/main/webapp/script.js
Comment (josh8551021) [reply portfolio/src/main/webapp/script.js]: Done. Let me know if the method I used isn't sufficient.

**Review by mkmg — APPROVED** [review]
Nice work :)

**Pull request merged**
